### PR TITLE
Create Header Dashboard UI

### PR DESCRIPTION
**Pull request by DKnehi**
image

### CODE REVIEW

File: javascript-practice/package.json [hunk 1]
```diff
@@ -2,8 +2,8 @@
   "name": "my-project",
   "source": "src/pages/login.html",
   "scripts": {
-    "start": "parcel src/pages/login.html",
-    "build": "rm -rf .parcel-cache && parcel build src/pages/login.html"
+    "start": "parcel src/pages/dashboard.html",
```
Comment (phuocduong-agilityio) [inline javascript-practice/package.json:5]: How can you run login page?

File: javascript-practice/src/pages/dashboard.html [hunk 2]
```diff
@@ -1,11 +1,49 @@
...
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <link rel="stylesheet" href="../styles/main.css" />
+    <title>Dashboard</title>
+  </head>
+  <body>
+    <div class="container">
+      <header class="header-dashboard flex">
+        <div class="header-dashboard-name-box flex-column">
+          <h2 class="header-dashboard-name">Hello, Dat</h2>
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/pages/dashboard.html]: For fake data, make it common. Update avatar also pls. In real project, we do not accept show private information on our application by fake data

File: javascript-practice/src/styles/layouts/header.css [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  color: var(--color-boulder);
+  letter-spacing: 0.25px;
+}
+
+.header-dashboard {
+  justify-content: space-between;
+  padding: 31px 23px 25px 43px;
+}
+
+.header-dashboard-profile-nitifi-box {
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/styles/layouts/header.css]: Typo

File: javascript-practice/src/js/views/userView.js [hunk 4]
```diff
@@ -1,59 +1,77 @@
...
+    this.isArrowUp = true
+    this.binDropDow()
   }
 
   bindFormLogin = (submitLogin) => {
     this.loginFormEl.addEventListener('submit', (e) => {
-      e.preventDefault();
-      const valueEmail = this.emailFormEl.value;
-      const valuePassword = this.passwordFormEl.value;
+      e.preventDefault()
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/views/userView.js]: Check code style pls

File: javascript-practice/src/js/views/userView.js [hunk 5]
```diff
@@ -1,59 +1,77 @@
...
+      this.emailErrorEl.textContent = ''
     })
     this.passwordFormEl.addEventListener('input', () => {
-      this.passwordErrorEl.textContent = '';
+      this.passwordErrorEl.textContent = ''
     })
-  };
+  }
+
+  binDropDow = () => {
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/views/userView.js]: First, typo
Second, function isn't clear what it does

File: javascript-practice/src/styles/layouts/header.css [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  line-height: 30px;
+  color: var(--color-black);
+  letter-spacing: 0.15px;
+}
+
+.header-dashboard-sub-name {
+  font-size: var(--font-size-small);
+  font-weight: var(--font-weight-light);
+  line-height: 21px;
+  color: var(--color-boulder);
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/styles/layouts/header.css]: Typo

File: javascript-practice/package.json (hunk 1 above)
Comment (DKnehi) [reply javascript-practice/package.json:5]: I am getting an error in building parcel. I'm looking for a fix

File: javascript-practice/src/pages/dashboard.html (hunk 2 above)
Comment (DKnehi) [reply javascript-practice/src/pages/dashboard.html]: I'll update it in the next task

File: javascript-practice/package.json (hunk 1 above)
Comment (phuocduong-agilityio) [reply javascript-practice/package.json:5]: With your solution, we can not run login page. I can accept for now. But please resolve it soon. We can not demo without login page

File: javascript-practice/src/js/views/userView.js [hunk 7]
```diff
@@ -28,32 +33,47 @@ export default class UserView {
 
       if (!valueEmail) {
         this.emailErrorEl.textContent = `${REQUIRED_FIELD_EMAIL}`;
-      }
+      };
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/views/userView.js]: No ; for block { }.

File: javascript-practice/src/js/views/userView.js [hunk 8]
```diff
@@ -28,32 +33,47 @@ export default class UserView {
...
       this.passwordErrorEl.textContent = '';
+    });
+  };
+
+  //After clicking on the arrow in the header, Logout will drop down
+  bindLogoutDropDow = () => {
+    this.arrowEl.addEventListener('click', () => {
+      if (this.isArrowUp) {
+        this.logoutEl.style.display = 'block';
+        this.arrowEl.style.transform = 'rotate(180deg)';
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/views/userView.js]: Do not style by JS. Should use CSS for styling

File: javascript-practice/src/pages/dashboard.html (hunk 2 above)
Comment (phuocduong-agilityio) [reply javascript-practice/src/pages/dashboard.html]: Use common data pls. Like Hello User

File: javascript-practice/src/js/views/userView.js [hunk 9]
```diff
@@ -47,13 +52,28 @@ export default class UserView {
...
     this.passwordFormEl.addEventListener('input', () => {
       this.passwordErrorEl.textContent = '';
+    });
+  };
+
+  //After clicking on the arrow in the header, Logout will drop down
+  bindLogoutDropDow = () => {
+    this.arrowEl.addEventListener('click', () => {
+      if (this.isArrowUp) {
+        this.logoutEl.style.display = 'block';
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/js/views/userView.js]: How about this one? Could you use class for styling?

File: javascript-practice/src/pages/dashboard.html [hunk 10]
```diff
@@ -1,11 +1,53 @@
...
+    <link rel="stylesheet" href="../styles/main.css" />
+    <title>Dashboard</title>
+  </head>
+  <body>
+    <div class="container">
+      <header class="header-dashboard flex">
+        <div class="header-dashboard-name-box flex-column">
+          <div class="flex header-dashboard-hi-name-box">
+            <h2 class="header-dashboard-name">Hello,</h2>
+            <!-- this is fake data. I'll update it in the next task -->
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/pages/dashboard.html]: Comment is not correct. Use TODO: https://www.jetbrains.com/help/webstorm/using-todo.html#create_todo

File: javascript-practice/src/styles/layouts/header.css [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+
+.header-dashboard-profile-sub-name {
+  font-weight: var(--font-weight-light);
+  font-size: var(--font-size-smallest);
+  color: var(--color-black);
+  line-height: 18px;
+  letter-spacing: 0.4px;
+}
+
+.arrow {
```
Comment (phuocduong-agilityio) [inline javascript-practice/src/styles/layouts/header.css:69]: Gather style for arrow

**Pull request merged**